noisy-simulator/src/density_matrix_simulator.rs

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! This module contains two structs: the `DensityMatrixSimulator` and its
+//! internal `DensityMatrix` state.
+
+#[cfg(test)]
+mod tests;
+
+use crate::{
+    handle_error, instrument::Instrument, kernel::apply_kernel, operation::Operation,
+    ComplexVector, Error, SquareMatrix, TOLERANCE,
+};
+use num_complex::Complex;
+
+/// A vectorized density matrix.
+#[derive(Debug, Clone)]
+pub struct DensityMatrix {
+    /// Dimension of the matrix. E.g.: If the matrix is 5 x 5, then dim is 5.
+    dim: usize,
```

Comment:
Should we rename this to dimension as other variable names aren't shorted?

Reply:
Yes, you are right.